src/com/amazon/ion/impl/IonReaderContinuableCore.java

```diff
@@ -0,0 +1,253 @@
+package com.amazon.ion.impl;
+
+import com.amazon.ion.Decimal;
+import com.amazon.ion.IntegerSize;
+import com.amazon.ion.IonCursor;
+import com.amazon.ion.IonInt;
+import com.amazon.ion.IonType;
+import com.amazon.ion.IvmNotificationConsumer;
+import com.amazon.ion.Timestamp;
+import com.amazon.ion.UnknownSymbolException;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Date;
+
+/**
+ * IonCursor with the core IonReader interface methods. Useful for adapting an IonCursor implementation into a
+ * system-level IonReader.
+ */
+interface IonReaderContinuableCore extends IonCursor {
+
+    /**
+     * Returns the depth into the Ion value that this reader has traversed.
+     * At top level the depth is 0.
+     */
+    int getDepth();
+
+    /**
+     * Returns the type of the current value, or null if there is no
+     * current value.
+     */
+    IonType getType();
+
+    /**
+     * Returns an {@link IntegerSize} representing the smallest-possible
+     * Java type of the Ion {@code int} at the current value.
+     *
+     * If the current value is {@code null.int} or is not an Ion
+     * {@code int}, or if there is no current value, {@code null} will
+     * be returned.
+     *
+     * @see IonInt#getIntegerSize()
+     */
+    IntegerSize getIntegerSize();
+
+    /**
+     * Determines whether the current value is a null Ion value of any type
+     * (for example, <code>null</code> or <code>null.int</code>).
+     * It should be called before
+     * calling getters that return value types (int, long, boolean,
+     * double).
+     */
+    boolean isNullValue();
+
+    /**
+     * Determines whether this reader is currently traversing the fields of an
+     * Ion struct. It returns false if the iteration
+     * is in a list, a sexp, or a datagram.
+     */
+    boolean isInStruct();
+
+    /**
+     * Gets the symbol ID of the field name attached to the current value.
+     * <p>
+     * <b>This is an "expert method": correct use requires deep understanding
+     * of the Ion binary format. You almost certainly don't want to use it.</b>
+     *
+     * @return the symbol ID of the field name, if the current value is a
+     * field within a struct.
+     * If the current value is not a field, or if the symbol ID cannot be
+     * determined, this method returns a value <em>less than one</em>.
+     *
+     */
+    @Deprecated
```

Comment:
It feels strange to me to see `@Deprecated` in a new interface. Are we sure we're doing this right? Or is a cast to a different interface the appropriate way to expose this?

Reply:
This is here for compatibility with the IonReader interface, where the method is marked `@Deprecated`. If you know a better way, let me know.